db/operations.go

```diff
@@ -73,6 +75,25 @@ func (o *Operation) mergeData(newData map[string]string) error {
 	return nil
 }
 
+func (o *Operation) getValues() ([]any, error) {
+	columns := make([]string, len(o.data))
+	i := 0
+	for column := range o.data {
+		columns[i] = column
+		i++
+	}
+	sort.Strings(columns)
+	values := make([]any, len(o.data))
+	for i, v := range columns {
+		convertedType, err := convertToType(o.data[v], o.table.columnsByName[v].scanType)
+		if err != nil {
+			return nil, fmt.Errorf("converting value %q to type %q: %w", o.data[v], o.table.columnsByName[v].scanType, err)
+		}
+		values[i] = convertedType
+	}
+	return values, nil
+}
+
 func (o *Operation) query(d dialect) (string, error) {
 	var columns, values []string
 	if o.opType == OperationTypeInsert || o.opType == OperationTypeUpdate {
```

Comment:
It's interesting to note now that `getValues()` is used exclusively by `Clickhouse` and `query(...)` only by `Postgres`.

It appears to me the abstraction about `getValues` and `query` should be moved to the dialect itself. It seems the way the queries are created and performed are simply different.

Moving `query` brings with it `prepareColValues`, `getPrimaryKeyWhereClause`, `normalizeValueType` and a few others.

Now that `Flush` is part of the dialect itself, those can be moved there.

Reply:
Done